tests/test.c

```diff
@@ -1990,6 +1990,44 @@ static void test_netrc(void)
 	rmdir(".test");
 }
 
+static void test_robots(void)
+{
+	ROBOTS *robots;
+	ROBOTS_PATH *path;
+	const char *sitemap;
+	const char *data =
+					"User-agent: *\n"\
+					"Disallow: /maps/reserve/terms\n"\
+					"Disallow: /maps/reserve/m/\n"\
+					"Disallow: /maps/reserve/b/\n"\
+					"Disallow: /about/views/\n"\
+					"Disallow: /intl/*/about/views/\n"\
+					"Sitemap: http://www.gstatic.com/s2/sitemaps/profiles-sitemap.xml\n"\
+					"Sitemap: https://www.google.com/sitemap.xml";
+
+	robots = wget_robots_parse(data, PACKAGE_NAME);
+
+	for (int it = 0; it < wget_vector_size(robots->paths); it++) {
+		if (!(path = wget_vector_get(robots->paths, it))) {
+			info_printf("Failed to get path '%s'\n from robots", path->path);
+			failed++;
+		} else {
+			ok++;
+		}
+	}
+
+	for (int it = 0; it < wget_vector_size(robots->sitemaps); it++) {
+		if (!(sitemap = wget_vector_get(robots->sitemaps, it))) {
```

Comment:
This is always true, you just iterate over all found sitemaps. You want to compare them with an expected array/list of sitemaps.